### PR TITLE
Generic Hash HLD

### PR DESCRIPTION
**Pull request by nazariig**
This document provides general information about GH implementation in SONiC
This scope of this HLD is:
* In scope:
   ECMP/LAG switch hash configuration

* Out of scope:
   ECMP/LAG switch hash seed configuration
   ECMP/LAG switch hash algorithm configuration

| PR title                                                                  | state                                                                                                                  | context                                                                                                                     |
| ------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------- |
| [GH YANG Model](https://github.com/sonic-net/sonic-buildimage/pull/13253) | GitHub issue/pull request detail | GitHub pull request check contexts |
| [GH YANG UM](https://github.com/sonic-net/sonic-buildimage/)              | GitHub issue/pull request detail      | GitHub pull request check contexts      |
| [GH DB Schema](https://github.com/sonic-net/sonic-swss-common/pull/733)   | GitHub issue/pull request detail  | GitHub pull request check contexts  |
| [GH VS Library](https://github.com/sonic-net/sonic-sairedis/pull/1192)    | GitHub issue/pull request detail    | GitHub pull request check contexts    |
| [GH Switch OA](https://github.com/sonic-net/sonic-swss/pull/2598)         | GitHub issue/pull request detail        | GitHub pull request check contexts        |
| [GH CLI](https://github.com/sonic-net/sonic-utilities/pull/2580)          | GitHub issue/pull request detail   | GitHub pull request check contexts   |
| [GH CLI UM](https://github.com/sonic-net/sonic-utilities/)                | GitHub issue/pull request detail       | GitHub pull request check contexts       |
| [GH Test Plan](https://github.com/sonic-net/sonic-mgmt/)                  | GitHub issue/pull request detail            | GitHub pull request check contexts            |
| [GH Test](https://github.com/sonic-net/sonic-mgmt/)                       | GitHub issue/pull request detail            | GitHub pull request check contexts            |

Signed-off-by: Nazarii Hnydyn [nazariig@nvidia.com](mailto:nazariig@nvidia.com)

### CODE REVIEW

**Comment by rlhui** [comment]
please look into how we can add api to expose what fields are actually being used in the hash by the switch.

**Comment by liat-grozovik** [comment]
@prsunny @rlhui kindly reminder to review. the follow up HLD review with the community is set for 12/06. 
Appreciate if you can review it following the prev review feedback

**Comment by nazariig** [comment]
> Does the hash field configuration apply for original incoming packets of the switch or final modified packets of the switch?

@rlhui to the original packets only - configuration is tunnel agnostic according to the SAI behavioral model

**Comment by nazariig** [comment]
> please look into how we can add api to expose what fields are actually being used in the hash by the switch.

done

File: doc/hash/hash-design.md [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+                "INNER_SRC_IP",
+                "INNER_L4_DST_PORT",
+                "INNER_L4_SRC_PORT"
+            ]
+        }
+    }
+}
+```
+
+### 2.4.4 Initial configuration
```
[the remark this answers is not in the document]
Comment (nazariig) [reply doc/hash/hash-design.md]: @prsunny just a kind reminder

**Comment by liat-grozovik** [comment]
Following the community review it was agreed to move forward with this HLD and have a second phase with new SAI API.
As of that, if any additional feedback to this HLD is available please send the sooner the better. Otherwise we will assume all is agreed and we will merge this HLD in the next few days.

**Comment by liat-grozovik** [comment]
@zhangyanzhao the HLD was reviewed and then we had a follow up. Code PRs are ready.
We should move forward with approval and merge as no feedback provided since the last review which was about month ago.

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
The feature and its current content is expected to be  in 202305 and under discussion if to take into 202211 as originally aimed.

**Pull request merged**